File: src/main/java/org/example/commands/ACommand.java
package org.example.commands;

import org.example.exception.ArgumentError;
import org.example.exception.ArgumentCountError;
import org.example.exception.ScriptExecutionError;

import java.io.FileNotFoundException;
/**
 * Абстрактный класс для команд
 */
public abstract class ACommand {
    private String name;
    private String info;
    private Integer countsArgument;
    public ACommand(String name, String info, Integer countsArgument) {
        this.name = name;
        this.info = info;
        this.countsArgument = countsArgument;
    }
    /**
     * Метод для проверки количества аргументов
     * @param args аргументы
     * @throws ArgumentCountError если количество аргументов не совпадает
     */
    public void valideCountsArgument(String[] args) throws ArgumentCountError {
        if (args.length!=countsArgument) {throw new ArgumentCountError(countsArgument,args.length);}
    }
    /**
     * Метод для выполнения команды
     * @param args аргументы
     * @throws ArgumentCountError если количество аргументов не совпадает
     * @throws ScriptExecutionError если произошла ошибка во время выполнения скрипта
     * @throws ArgumentError если аргументы не корректны
     */
    public abstract void execute(String[] args) throws ArgumentCountError, ScriptExecutionError, ArgumentError;

    @Override
    public String toString() {
        return this.name + ": " + this.info;
    }
}

File: src/main/kotlin/commands/ACommand.kt
package org.chousik.commands

import exeption.ArgumentCountError


abstract class ACommand(private val nameValue: String, private val info: String, private val countsArgument: Int) {
    val name: String
        get() {
            return nameValue
        }

    fun validCountsArgument(args: Array<String>) {
        if (args.size != countsArgument) {
            throw ArgumentCountError(countsArgument, args.size)
        }
    }

    abstract fun execute(args: Array<String>)

    override fun toString(): String {
        return this.nameValue + ": " + this.info
    }
}